onadata/apps/api/permissions.py

```diff
@@ -535,10 +537,14 @@ class IsAuthenticatedSubmission(BasePermission):
     """
 
     def has_permission(self, request, view):
-        username = view.kwargs.get("username")
         form_pk = view.kwargs.get("xform_pk")
-        project_pk = view.kwargs.get("project_pk")
-        if request.method in ["HEAD", "POST"] and request.user.is_anonymous:
+        if request.method == "HEAD" and request.path.endswith(
+            f"enketo/{form_pk}/submission"
+        ):
+            return True
+        elif request.method in ["HEAD", "POST"] and request.user.is_anonymous:
```

Comment:
Could we check that the request.user is authenticated? It is by passing permissions checks in the section below. perhaps what needs to be revisited is lines 561 - 564:

```python
...
    if profile.require_auth:
       return False
...
```

I wonder how does it affect head requests only and not post requests as well?

Reply:
@FrankApiyo could you add some test(s) - it could provide better clarity on the issue that is being addressed.